usecases/scenario_testrun_usecase.go

```diff
@@ -27,6 +28,27 @@ func (usecase *ScenarioTestRunUsecase) ActivateScenarioTestRun(ctx context.Conte
 	if err := usecase.enforceSecurity.CreateTestRun(organizationId); err != nil {
 		return models.ScenarioTestRun{}, err
 	}
+	indexesToCreate, _, err := usecase.clientDbIndexEditor.GetIndexesToCreate(
+		ctx,
+		organizationId,
+		input.ScenarioIterationId,
+	)
+	if err != nil {
+		return models.ScenarioTestRun{}, errors.Wrap(err,
+			"Error while fetching indexes to create in ActivateScenarioTestRun")
+	}
+	if len(indexesToCreate) > 0 {
+		return models.ScenarioTestRun{}, errors.Wrap(
+			models.ErrScenarioIterationRequiresPreparation,
+			fmt.Sprintf("Cannot activate the scenario in test mode: it requires data preparation to be run first for %d indexes", len(indexesToCreate)),
+		)
+	}
+
+	errIdx := usecase.clientDbIndexEditor.CreateIndexesAsync(ctx, organizationId, indexesToCreate)
```

Comment:
We have a bit of a problem here, with this implementation: the method is async (as its name suggests) so it doesn't wait for the index creation to be done to return.
However, we don't wont the test run execution to start before the indexes are finished creating.
2-3 options:
- all of the test run creation becomes async, and there is no immediate user feedback (other than "ok taken into account) when he clicks "start test run" => definitely bad
- we create the test run in an intermediate status, and update its status to "active" after the index creation is finished => much better IMO
- we take inspiration from the way it's currently done in the builder: we require the indexes to be present before the user can click "activate" => better than 1 but I don't think it's the solution here. Pretty sure we should go for 2.

Reply:
if we do not want to execute the test run before the indexes has been created, we should make it blocking.

Reply:
ideally we should make it blocking but given the way it has been written, I do not see how we can make sure the indexes have been created correctly before doing anything else. unless we add a new feature to do so

Reply:
IMO, we need call a method that will: set the test run to an intermediate "preparation pending" (or such) status, starts the index creation, rolls back the status if there is an error or updates the status to "running" if the index creation succeeds. The easiest way of doing that would probable be to have another method like "createIndexesAsync" that accepts a pair of callbacks. But keep it simple, we'll touch this again very soon anyway.

Making it blocking isn't much of a solution because we can't just wait for the endpoint to respond for more than seconds.